src/PowerShellEditorServices.Host/EditorServicesHost.cs

```diff
@@ -463,7 +463,19 @@ private IServerListener CreateServiceListener(MessageProtocolType protocol, Edit
 
                 case EditorServiceTransportType.NamedPipe:
                 {
-                    return new NamedPipeServerListener(protocol, config.Endpoint, this.logger);
+                    string endpoint = config.Endpoint;
+                    int splitIndex = endpoint.IndexOf(';');

```

Comment:
I'm more inclined to say that the character that separates the two file names should be the `DirectorySeparator` (`System.IO.Path.DirectorySeparatorChar`) since `;` could be a part of a valid file name. I don't think `GetRandomFileName` would ever spit out a `;` but at least if we use the `DirectorySeparatorChar` we can be sure.

Reply:
I spoke to @rjmholt about the use of `Endpoint` here. Historically, this was used to support TCP and NamedPipes but since we don't have TCP anymore, we could change this `EditorServiceTransportConfig` to have a `Pipe` and `WritePipe` property (but named better) that way you don't have to do this string splitting.

I think I'm ok if you don't want to do this in this PR - but it's pretty important so if you want to take it on I'd really appreciate it.